src/js/client/helpers.js

```diff
@@ -0,0 +1,179 @@
+ */
+Handlebars.registerHelper("lookupGenericPrefValue", function (prefsSet, commonTerm) {
+    return editPrefs.lookupGenericPrefValue(prefsSet, commonTerm);
+});
+
+/**
+ * Lookup Application Specific Setting. Similar to `lookupGenericPrefValue` lookups
+ * in regard to return values, but takes an extra argument for the product.
+ */
+Handlebars.registerHelper("lookupProductPrefValue", function (prefsSet, product, settingTerm) {
+    return editPrefs.lookupProductPrefValue(prefsSet, product, settingTerm);
+});
+
+/**
+ * appIDfromURI Handlebars Helper
+ *
+ * Extract a GPII Application ID from a preferences URI.
+ * ie. Convert http://registry.gpii.net/applications/org.gnome.orca
+ * to org.gnome.orca .
+ * This should likely be replaced with correct ontology usage.
+ *
+ * @param {String} uri - a flat URI of a GPII application.
+ * @return {String} The application ID.
+ */
+Handlebars.registerHelper("appIDfromURI", function (uri) {
+    return uri.split(/\//).pop();
+});
+
+/**
+ * appURIfromID Handlebars Helper
+ *
+ * Extract a GPII Flat URL ID from a  URI.
+ * ie. Convert org.gnome.orca to
+ * http://registry.gpii.net/applications/org.gnome.orca
+ * This should likely be replaced with correct ontology usage.
+ *
+ * @param {String} uri - The application ID.
+ * @return {String} A flat URI of a GPII application.
+ */
+Handlebars.registerHelper("appURIfromID", function (appId) {
+    return "http://registry.gpii.net/applications/" + appId;
+});
+
+/**
+ * checkForApp Handlebars helper
+ * Takes a list of GPII Applications, and returns the object if
+ * it's of type appId. It would be nice to wrap some of the infusion
+ * find/map functions as helpers rather than use this custom function.
+ *
+ * @param {Object} prefsSafeApplications - The listing of GPII app solutions.
+ * @param {String} appId - The ID of the application we're looking for.
+ * @return {Object} The solution entry of the IP, otherwise undefined.
+ */
+Handlebars.registerHelper("checkForApp", function (prefsSafeApplications, appId) {
+    return fluid.find(prefsSafeApplications, function (i) {
+        if (i.appId === appId) {
+            return i;
+        }
+    });
+});
+
+/**
+ * checkForSupportedSetting Handlebars helper
+ * Takes a solutions registry entry and settingKey, then checks to see if
+ * it's a setting that the solution supports. This is primarily used to see
+ * if a generic preference setting is supported by the solution and used
+ * in one of it's transformations.
+ * The underlying functionality for this should be moved to a utility method
+ * with unit tests.
+ *
+ * @param {Object} solution - The solution registry entry.
+ * @param {String} settingKey - The generic pref setting key.
+ * @return {Boolean} Returns `true` if this solution supports the setting, otherwise `false`.
+ */
+Handlebars.registerHelper("checkForSupportedSetting", function (solution, settingKey) {
+    var togo = false;
+    fluid.each(solution.settingsHandlers, function (i) {
```

Comment:
This can be two nested fluid.find rather than 2*each and a closure-busting assignment

Reply:
Actually, this bit of work is waiting on future input from UX (with big TODO's around where this helper was used) so I've removed this bit for now, and the corresponding bit of template.